File: 프로그래머스/lv2/131704. 택배상자/택배상자.py
def solution(order):
    answer = 0
    stack = []
    l = len(order)
    i=1
    while i <= l:
        stack.append(i)
        while stack and stack[-1] == order[answer]:
            answer+=1
            stack.pop()
        
        i+=1
    return answer

File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;

class Solution {
    public int solution(int[] order) {
        int answer = 0;
        
        Stack<Integer> stack = new Stack<>();
        // 커서 역할
        int idx = 0;
            
        for (int i=1; i<=order.length; i++) {
            
            if(order[idx] != i) {
                stack.push(i);
            } else {
                idx++;
                answer++;
            }
            while(!stack.isEmpty() && stack.peek() == order[idx]) {
                stack.pop();
                idx++;
                answer++;
            }
        }
        
        return answer;
    }
}